com.unity.shadergraph/Editor/Importers/ShaderGraphImporter.cs

```diff
@@ -115,7 +115,32 @@ public override void OnImportAsset(AssetImportContext ctx)
 #endif
             {
                 var text = GetShaderText(path, out configuredTextures, sourceAssetDependencyPaths, graph);
-                shader = ShaderUtil.CreateShaderAsset(text, false);
+
+#if UNITY_2021_1_OR_NEWER
+                // 2021.1 or later is guaranteed to have the new version of this function
+                shader = ShaderUtil.CreateShaderAsset(ctx, text, false);
+#else
+                // earlier builds of Unity may or may not have it
+                // here we try to invoke the new version via reflection
+                var createShaderAssetMethod = typeof(ShaderUtil).GetMethod(
+                    "CreateShaderAsset",
+                    System.Reflection.BindingFlags.Static | System.Reflection.BindingFlags.Public | System.Reflection.BindingFlags.ExactBinding,
+                    null,
+                    new Type[] { typeof(AssetImportContext), typeof(string), typeof(bool) },
+                    null);
+
+                if (createShaderAssetMethod != null)
```

Comment:
Ok so we can get rid of the reflection code here?

Reply:
I guess the reflection is needed in the 2020.2+ version of the package only to avoid some pain with users with Unity alpha getting the latest shadergraph. Basically without the reflection the latest shadergraph will fail compiling unless they update their Unity to 2020.2b1. Personally I'd be fine having that bit of alpha/beta update pain for cleaner code, as no users should stick into an alpha version for too long anyway. Or maybe have the reflection here for a while to allow painless alpha->beta transition and then on later date clean the code up.

However, for versions of shadergraph that work with older Unity versions the reflection path is needed permanently as we're backporting the API addition and want to provide the dependency fix there, but we can't force the users to upgrade to latest patch release.

Reply:
Yeah, plan is to remove it once we don't think many users will be hitting it anymore, probably near end of beta.  In any case once we jump to 2021.1 it will remove itself.   And yeah, the backports will need the code.